### PR TITLE
Add round robin replies for A and AAAA replies

### PR DESCRIPTION
**Pull request by miekg**

### CODE REVIEW

**Comment by crosbymichael** [comment]
Is this going to cause issues with sessions if this is the default and cannot be turned off?

**Comment by miekg** [comment]
I was contemplating an option. What is a session in this context?

Note that you can never depend on the RR order in an dns message.

**Comment by miekg** [comment]
Add a -noroundrobin option to skydns.

File: main.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func init() {
 	flag.DurationVar(&wtimeout, "wtimeout", 2*time.Second, "Write timeout")
 	flag.StringVar(&secret, "secret", "", "Shared secret for use with http api")
 	flag.StringVar(&nameserver, "nameserver", "", "Nameserver address to forward (non-local) queries to e.g. 8.8.8.8:53,8.8.4.4:53")
+	flag.BoolVar(&norr, "noroundrobin", false, "Do not round robin A/AAAA replies")
```
Comment (crosbymichael) [inline main.go]: Do you think this should default to true and rename to `enable-round-robin`?

**Comment by bketelsen** [comment]
i like defaulting it.

File: main.go (hunk 1 above)
Comment (miekg) [inline main.go]: Why would the string-sorted order for IP addresses be better than random?

**Comment by miekg** [comment]
Tot clarify: defaulting to... Round robin?

Would a better idea be to load balance according to weight and prio, if we
have that info? Just a thought.

**Comment by miekg** [comment]
But that would be much much harder as we need a history of served replies.

File: main.go (hunk 1 above)
Comment (crosbymichael) [inline main.go]: I see your point

**Comment by crosbymichael** [comment]
LGTM

**Comment by bketelsen** [comment]
LGTM

**Comment by miekg** [comment]
rebase, merge, rebase, edit. Should be in good shape now.

**Pull request merged**